cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/SnapshotDataset.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib;
+
+import co.cask.cdap.api.data.batch.RecordScanner;
+import co.cask.cdap.api.data.batch.Split;
+import co.cask.cdap.api.data.batch.SplitReader;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.dataset.table.Delete;
+import co.cask.cdap.api.dataset.table.Get;
+import co.cask.cdap.api.dataset.table.Increment;
+import co.cask.cdap.api.dataset.table.Put;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scan;
+import co.cask.cdap.api.dataset.table.Scanner;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.tephra.Transaction;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ *  Dataset client for snapshot datasets.
+ */
+public class SnapshotDataset extends AbstractDataset implements Table {
+
+  private static final byte[] METADATA_ROW_KEY = { 'v', 'e', 'r', 's', 'i', 'o', 'n' };
+  private static final byte[] METADATA_KEY_COLUMN = { 'v', 'a', 'l', 'u', 'e' };
+
+  private Table metadataTable;
+  private Table mainTable;
+  private Transaction tx;
+
+  public SnapshotDataset(String name, Table metadataTable, Table mainTable) {
+    super(name, metadataTable, mainTable);
+    this.metadataTable = metadataTable;
+    this.mainTable = mainTable;
+  }
+
+  public Long getCurrentVersion() {
+    return metadataTable.get(METADATA_ROW_KEY).getLong(METADATA_KEY_COLUMN);
+  }
+
+  public long getTransactionId() {
+    return tx.getTransactionId();
+  }
+
+  public void updateMetaDataTable(long newVersion) {
```

Comment:
Shouldn't be public. The maintenance of the meta information for this Dataset should be transparent to user of this Dataset.